src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -9336,11 +9336,23 @@ private TypeWithState InferResultNullabilityOfBinaryLogicalOperator(BoundExpress
             var operand = node.Operand;
             var typeExpr = node.TargetType;
 
-            VisitRvalue(operand);
+            VisitPossibleConditionalAccess(operand, out var conditionalStateWhenNotNull);
+            Unsplit();
+
+            LocalState stateWhenNotNull;
+            if (!conditionalStateWhenNotNull.IsConditionalState)
+            {
+                stateWhenNotNull = conditionalStateWhenNotNull.State;
+            }
+            else
+            {
+                stateWhenNotNull = conditionalStateWhenNotNull.StateWhenTrue;
+                Join(ref stateWhenNotNull, ref conditionalStateWhenNotNull.StateWhenFalse);
+            }
+
             Debug.Assert(node.Type.SpecialType == SpecialType.System_Boolean);
 
-            Split();
-            LearnFromNonNullTest(operand, ref StateWhenTrue);
```

Comment:
I would have expected some negative impact from removing this call, at least when the operand isn't a conditional access. Am I missing something?

Reply:
It's baked in to the VisitPossibleConditionalAccess. We effectively do a "LearnFromNonNullTest" on the operand and update the stateWhenNotNull even if the operand isn't a conditional access.

Reply:
Ah, I see it. Thanks!